wagtail/admin/templates/wagtailadmin/pages/search.html

```diff
@@ -19,8 +11,8 @@
 
 {% block content %}
     {% trans "Search" as search_str %}
-    {% include "wagtailadmin/shared/header.html" with title=search_str search_url="wagtailadmin_pages:search_results" icon="search" %}
-
+    {% url 'wagtailadmin_pages:search' as search_results_url %}
+    {% include "wagtailadmin/shared/header.html" with  with title=search_str search_url="wagtailadmin_pages:search_results" icon="search" search_results_url=search_results_url search_target="#page-results" %}
```

Comment:
```suggestion
    {% include "wagtailadmin/shared/header.html" with title=search_str search_url="wagtailadmin_pages:search_results" icon="search" search_results_url=search_results_url search_target="#page-results" %}
```

Reply:
Thanks

Reply:
@lb- took this for a spin and a few things look broken: https://www.loom.com/share/1973dfed448d421992faef126ca484b1?sid=51fbf8e9-9002-4e15-94df-daf11d36e9b7

Reply:
Thanks, dodgy rebase sorry, I did not see the change to the URL name `wagtailadmin_pages:search_results` has now been adopted.

![Screenshot 2023-07-12 at 10 10 50 pm](https://github.com/wagtail/wagtail/assets/1396140/ae281350-3f08-4e35-a838-880e219327c0)

Reply:
I also found a bug on this page, caused by d554cbe3103028094e095b4512323ca2967aca4a 

There is an invisible submit button that's meant to submit a search (loads full page) for screen readers, as of that commit it now returns the partial HTML response.

I have fixed this in this PR also.

Reply:
ooh, nice!

The changes work well for me, functionally. While at it, added a note to update wagtailmedia

Reply:
Thanks @zerolab - I have also updated the upgrade considerations to account for usage of the shared header template.